src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2071,19 +2069,21 @@ class CgroupUtils:
                 pbs.logmsg(pbs.EVENT_DEBUG2,
                            "%s: Job %s contains defunct process" %
                            (caller_name(), jobid))
-                raise CgroupProcessingError("Job contains defunct process")
+                # Use a list comprehension to remove all instances of the
+                # number 1
+                pids = [x for x in pids if x != 1]
         # check pids to make sure that they are owned by the job owner
         if pbs_version[0] >= 13:
```

Comment:
It does get ugly fast, but we have early adopters that want the latest version of the hook regardless of PBS version they are running. While qsub is a compiled command, hooks are meant to be modifiable. In general, hooks should work across a wide range of PBS versions unless development has deliberately broken compatibility.

Reply:
PBS hooks are considered core PBS functionality.  I still think this means if a new version of the hook comes out, it will depend on the version that it was shipped with.  We won't go out of our way to break it, but breakage can happen.  An example is that you usually can use a newer version of the scheduler with an older server.  We don't go out of our way to break that communication, but some times the communication does break.  To fix this, PBS has to be upgraded as a whole.  We don't have anything built into that communication pathway to allow for this.  I don't think we should do it in hooks either.  Not to mention that this is an OSS checkin and the earliest version of OSS is 14.1.  You're checking for >= 13.